Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -1741,6 +1743,8 @@ NonnullRefPtr<ObjectExpression> Parser::parse_object_expression()
                     syntax_error("Property name '__proto__' must not appear more than once in object literal");
                 has_direct_proto_property = true;
             }
+            if (is_proto && property_type == ObjectProperty::Type::KeyValue)
+                property_type = ObjectProperty::Type::ProtoSetter;
```

Comment:
It looks like the check above should also only happen for `ObjectProperty::Type::KeyValue` so it might be possible to combine the checks a little more. But it's fine to leave it like this.

Reply:
For example: `({ __proto__: null, get __proto__() {} });` is valid but probably not for us right now (don't have access to test this at the moment)

Reply:
```js
const x = ({ __proto__: null, get __proto__() { return 1; } });
console.log(x["__proto__"]);
console.log(Object.getPrototypeOf(x));
```
Prints
```
1
null
```
on both LibJS and v8

Reply:
Ah alright, nice.
In that case definitely not needed, thanks for looking into it.